File: RandomTeamChooserCs/TeamTool.cs
using System;
using System.Collections.Generic;

namespace RandomTeamChooser
{
    public static class TeamTool
    {
        private static List<string> activePlayers = new List<string>();
        private static Random rnd = new Random();

        private static float teamsCount = 0;
        private static float playersCount = 0;
        private static float playersPerTeamCount = 0;

        /*=================================
         *      MAIN PROCESS
         *=================================*/
        public static void GenerateTeams()
        {
            Console.Write("==== TEAM TOOL ====\n\n");

            CountTeams();

            CountPlayers();

            // Calculate player count equally for all teams
            playersPerTeamCount = (playersCount / teamsCount);

            Console.Write("\n");

            // If number is not a whole number, adjust the index count of list, because we can't add a "half" player to the team
            // HACK: Just count the length of string, if bigger than 1, its not a whole number (decimal). Basically adding a invisible player to make it work
            string stringToCheck = playersPerTeamCount.ToString();
            if (stringToCheck.Length > 1)
            {
                double newPlayerCount = Convert.ToDouble(playersPerTeamCount);
                newPlayerCount = Math.Round(playersPerTeamCount, 0, MidpointRounding.AwayFromZero);
                playersPerTeamCount = Convert.ToSingle(newPlayerCount);

                activePlayers.Add("");
            }

            // Save all active players
            for (int i = 0; i < playersCount; i++)
            {
                Console.Write(string.Format("Player {0}: ", i + 1));
                activePlayers.Add(Console.ReadLine());
            }

            Console.Write("\n");

            // Randomize player order
            //activePlayers = activePlayers.OrderBy(x => rnd.Next()).ToList();
            activePlayers = ShuffleList(activePlayers);

            for (int i = 0; i < teamsCount; i++)
            {
                Console.WriteLine(string.Format("==== TEAM {0} ====", i + 1));

                // Create array for each team (faster than list)
                string[] team = new string[Convert.ToInt32(playersPerTeamCount)];

                // Get players from shuffled list and put them in the current team
                for (int j = 0; j < playersPerTeamCount; j++)
                {
                    string player = GetPlayerFromList(activePlayers);

                    team[j] = player;

                    // Write the invisible player in the same line to hide the space between players
                    if (team[j] == "") Console.Write(team[j]);
                    else Console.WriteLine(team[j]);
                }
            }

            Console.WriteLine("\nPress enter to exit...");
            Console.ReadLine();
        }


        /*=================================
         *      HELPER FUNCTIONS
         *=================================*/

        // Set team count
        public static void CountTeams()
        {
            Console.Write("How many teams are playing? ");
            string teamsCountString = Console.ReadLine();
            if (!float.TryParse(teamsCountString, out teamsCount))
            {
                // Error handling
                Console.WriteLine("\nPlease type in a whole number...\n");
                CountTeams();
                return;
            }
            teamsCount = Convert.ToSingle(teamsCountString);
        }

        // Set player count
        public static void CountPlayers()
        {
            Console.Write("How many players are online? ");
            string playersCountString = Console.ReadLine();
            if (!float.TryParse(playersCountString, out playersCount))
            {
                // Error handling
                Console.WriteLine("\nPlease type in a whole number...\n");
                CountPlayers();
                return;
            }
            playersCount = Convert.ToSingle(playersCountString);
        }

        // Randomize the order of all elements in a list
        public static List<T> ShuffleList<T>(List<T> list)
        {
            for (int i = 0; i < list.Count; i++)
            {
                int index = rnd.Next(0, i);
                T value = list[index];
                list[index] = list[i];
                list[i] = value;
            }

            return list;
        }

        // Draw first player from the list and remove him
        public static string GetPlayerFromList(List<string> playerList)
        {
            string player = playerList[0];
            playerList.RemoveAt(0);
            return player;
        }
    }
}

File: RandomTeamChooserCpp/TeamTool.cpp
#include "TeamTool.h"

/*=================================
*      MAIN PROCESS
*=================================*/

void TeamTool::GenerateTeams()
{
    std::cout << "==== TEAM TOOL ====\n\n";

    CountTeams();

    CountPlayers();

    // Calculate player count equally for all teams
    playersPerTeamCount = (playersCount / teamsCount);

    std::cout << "\n";

    // If number is not a whole number, adjust the index count of list, because we can't add a "half" player to the team
    // HACK: Check if playersPerTeamCount is a whole number, if not, its a decimal. Basically adding a invisible player to make it work
    std::string stringToCheck = std::to_string(playersPerTeamCount);
    float result;
    result = atof(stringToCheck.c_str());

    if (result != static_cast<int>(result))
    {
        double newPlayerCount = playersPerTeamCount;
        newPlayerCount = ceil(playersPerTeamCount);
        playersPerTeamCount = newPlayerCount;

        activePlayers.push_back("");
    }

    // Save all active players
    for (int i = 0; i < playersCount; i++)
    {
        std::string inputString;

        std::cout << "Player " << i + 1 << ": ";
        std::cin >> inputString;
        activePlayers.push_back(inputString);
    }

    std::cout << "\n";

    // Randomize player order
    activePlayers = ShuffleList(activePlayers);

    for (int i = 0; i < teamsCount; i++)
    {
        std::cout << "==== TEAM " << i + 1 << " ====\n";

        // Create array for each team
        //int teamSizeInt = playersPerTeamCount;
        std::string team[100];

        // Get players from shuffled list and put them in the current team
        for (int j = 0; j < playersPerTeamCount; j++)
        {
            std::string player = GetPlayerFromList(activePlayers);

            team[j] = player;

            // Write the invisible player in the same line to hide the space between players
            if (team[j] == "") std::cout << team[j];
            else std::cout << team[j] << "\n";
        }
    }

    std::cout << "\nPress Enter to continue...";
    fgetc(stdin);
    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
}


/*=================================
*      HELPER FUNCTIONS
*=================================*/

// Set team count
void TeamTool::CountTeams()
{
    std::string teamsCountString;

    std::cout << "How many teams are playing? ";
    std::cin >> teamsCountString;

    float result;
    result = atof(teamsCountString.c_str());

    if (result == 0)
    {
        // Error handling
        std::cout << "\nPlease type in a whole number...\n\n";
        CountTeams();
        return;
    }

    teamsCount = std::stof(teamsCountString);
}

// Set player count
void TeamTool::CountPlayers()
{
    std::string playersCountString;

    std::cout << "How many players are online? ";
    std::cin >> playersCountString;

    float result;
    result = atof(playersCountString.c_str());

    if (result == 0)
    {
        // Error handling
        std::cout << "\nPlease type in a whole number...\n\n";
        CountPlayers();
        return;
    }

    playersCount = std::stof(playersCountString);
}

// Randomize the order of all elements in a list
std::vector<std::string> TeamTool::ShuffleList(std::vector<std::string> list)
{
    std::random_device rd;
    std::default_random_engine rng(rd());
    std::shuffle(std::begin(list), std::end(list), rng);
    return list;
}

// Draw first player from the list and remove him
std::string TeamTool::GetPlayerFromList(std::vector<std::string> &playerList)
{
    std::string player = playerList[0];
    playerList.erase(playerList.begin());
    return player;
}